### PR TITLE
[v7r0] [fixes out of hackhathon of 11/07] SiteDirectors send by default Pilot3, plus several other fixes

### PR DESCRIPTION
**Pull request by fstagni**
BEGINRELEASENOTES

*Interfaces
CHANGE: removed deprecated methods from Dirac.py

*WMS
NEW: SiteDirectors send by default Pilot3 (flag for pilot2 is still kept)
FIX: removed unicode_literals (messes up things with DEncode)

ENDRELEASENOTES

### CODE REVIEW

**Comment by andresailer** [comment]
You also have to get rid of the `from builtins import str, int` etc

**Comment by andresailer** [comment]
or maybe in DEncode
```
from builtins import str as newstr
from builtins import int as newint
...
g_dEncodeFunctions[type(newint(0))] = encodeInt
g_dEncodeFunctions[type(newstr('s'))] = encodeString
```

**Comment by chaen** [comment]
No, let's put the focus on that in another release. I propose to have v7r0 out, in v7r1 we do the first transition step to JSON, in v7r2 DISET goes away, allowing us to try that again

**Comment by fstagni** [comment]
I didn't remove the `from builtins` imports because they are used only for checking the type. Hopefully, it's fine.

**Comment by andresailer** [comment]
No, that is not fine.
In the exception the `future.types.new<Type>.new<Type>` is a clear indication that the import from `builtins` is a problem, as `from __future__` works without `future`.

```
In [1]: from __future__ import unicode_literals
In [2]: s = 'stra'
In [4]: type(s)
Out[4]: unicode
In [5]: s = str('2')
In [6]: type(s)
Out[6]: str
In [7]: from builtins import str
In [8]: s = str('2')
In [9]: type(s)
Out[9]: future.types.newstr.newstr
```

Thus adding the additional entries in the g_dEncodeFunctions would maybe allow us to keep using the `unicode_literals` and the `newstr`

**Comment by chaen** [comment]
Maybe. But I still think it's too much at once

**Comment by fstagni** [comment]
OK OK I will remove the imports from `builtins` then. It will be simpler for this release.

File: WorkloadManagementSystem/Service/JobMonitoringHandler.py [hunk 1]
```diff
@@ -289,16 +286,6 @@ def export_getJobLoggingInfo(jobID):
 
     return gJobLoggingDB.getJobLoggingInfo(jobID)
 
-##############################################################################
-  types_getJobsParameters = [list, list]
-
-  @staticmethod
-  @deprecated("Unused")
-  def export_getJobsParameters(jobIDs, parameters):
```
Comment (andresailer) [inline WorkloadManagementSystem/Service/JobMonitoringHandler.py]: In fact we are using this method in iLCDirac

PS: having the deprecated on the service side is not very useful, unless one reads the whole log...
```
NOTE: export_getJobsParameters function is deprecated (Unused).
```

File: Core/Utilities/Graphs/GraphData.py [hunk 2]
```diff
@@ -316,65 +318,65 @@ def truncateLabels( self, limit = 10 ):
...
 
-  def getStats( self ):
+  def getStats(self):
     """ Get statistics of the graph data
     """
 
-    numData = self.getPlotNumData( zipFlag = False )
-    if not numData:
+    numData = self.getPlotNumData(zipFlag=False)
+    if not len(numData):
```
Comment (andresailer) [inline Core/Utilities/Graphs/GraphData.py]: As pylint usually complains about this check, maybe add a comment, and the relevant `  # pylint: disable=len-as-condition`?

File: README.rst [hunk 3]
```diff
@@ -78,7 +78,9 @@ The contributions are subject to reviews.
 Pylint, and pep8 style checker are run regularly on the source code. The .pylintrc file defines the expected coding rules and peculiarities (e.g.: tabs consists of 2 spaces instead of 4).
 Each Pull Request is checked for pylint and pep8 compliance.
 
-Each PR is a also subject to check for python 3 compatibility. If you are issuing PRs, for each of the python files touched please run (and react to)::
+Each PR is a also subject to check for python 3 compatibility.
+If you are issuing PRs that are devoted to future versions of DIRAC (so, not for patch releases),
```
Comment (andresailer) [inline README.rst]: If you want the line break also in the rendered file, you need to have an empty line I think.

Comment (fstagni) [reply README.rst]: OK but I don't want it :)

**Pull request merged**